### PR TITLE
Renamed the Workflow.mco to Workflow.mco_model

### PR DESCRIPTION
**Pull request by Corwinpro**
This PR approaches #246 

### Summary

The attribute `mco` of the `Workflow` class refers to the `Instance(BaseMCOModel)`. There are such entities as `BaseMCO` alongside the `BaseMCOModel`, and therefore this naming is misleading. We approach to resolve this inconsistency.

### Done

- We rename the attribute to `Workflow.mco_model`, and change all implicit and explicit references to `mco` to `mco_model`.

### Compatibility

This change doesn't breaks the backwards compatibility for workflow json files but throws a deprecation warning. We suggest that all workflow json file `workflow["mco"]` elements must be renamed to `workflow["mco_model"]`. In addition, the `force-wfmanager` requests to `Workflow.mco` should be changed to `Workflow.mco_model`.

### CODE REVIEW

**Comment by flongford** [comment]
I think we will need to create a PR for the `force_wfmanager` too - when adding a new MCO in the WfManager, a message appears: 

    traits.trait_errors.TraitError: The 'mco' trait of a Workflow instance is 'read only'.

Ironically, this looks like the failing line:

    File "/Users/flongford/FORCE/force-wfmanager/force_wfmanager/ui/setup/workflow_view.py", line 122, in set_mco
        self.model.mco = mco_model

File: force_bdss/app/workflow_evaluator.py [hunk 1]
```diff
@@ -22,7 +25,7 @@ class WorkflowEvaluator(HasStrictTraits):
     workflow = Instance(Workflow)
 
     #: A reference to the mco model of the workflow instance.
-    mco_model = DelegatesTo('workflow', prefix='mco')
+    mco_model = DelegatesTo("workflow", prefix="mco_model")
```
Comment (flongford) [inline force_bdss/app/workflow_evaluator.py]:
```suggestion
    mco_model = DelegatesTo("workflow", prefix="_mco_model")
```

File: force_bdss/core/workflow.py [hunk 2]
```diff
@@ -12,11 +12,31 @@
...
+    #: Can be None if no MCOModel has been specified yet.
+    _mco_model = Instance(
+        BaseMCOModel, allow_none=True  # , transient=True, visible=False
+    )
+
+    #: Deprecated MCO attribute references to the mco_model attribute.
+    #: It is present to prevent possible dependencies API breaks.
+    mco = Property(depends_on="mco_model")  # , transient=True, visible=False)
+
+    mco_model = Property(depends_on="mco_model")
```
Comment (flongford) [inline force_bdss/core/workflow.py]:
```suggestion
    mco_model = Property(depends_on="_mco_model")
```

Comment (Corwinpro) [reply force_bdss/core/workflow.py]: Brilliant. Thank you.

**Pull request merged**